### PR TITLE
feat: clone sources and destinations API

### PR DESCRIPTION
**Pull request by harshithmullapudi**
## What
Now you can clone Source and destination connections

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java [hunk 1]
```diff
@@ -249,6 +272,16 @@ private DestinationRead buildDestinationRead(final UUID destinationId, final Con
     return toDestinationRead(dci, standardDestinationDefinition);
   }
 
+  private DestinationRead buildDestinationReadWithSecrets(final UUID destinationId)
+          throws ConfigNotFoundException, IOException, JsonValidationException {
+
+    // remove secrets from config before returning the read
+    final DestinationConnection dci = Jsons.clone(configRepository.getDestinationConnection(destinationId));
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java:279]: the object returned here will not have secrets.  but the method name says with secrets. do you mean to use this `getDestinationConnectionWithSecrets`?

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SourceHandler.java [hunk 2]
```diff
@@ -220,6 +244,15 @@ private SourceRead buildSourceRead(final UUID sourceId, final ConnectorSpecifica
     return toSourceRead(sourceConnection, standardSourceDefinition);
   }
 
+  private SourceRead buildSourceReadWithSecrets(final UUID sourceId)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    // read configuration from db
+    final SourceConnection sourceConnection = configRepository.getSourceConnection(sourceId);
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/SourceHandler.java:250]: same as above.

**Review by cgardens — CHANGES_REQUESTED** [review]
looks good! API looks solid. in the implementation, i'm not sure we'll actually copy the secrets over properly. see my comments below.

File: airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java (hunk 1 above)
Comment (harshithmullapudi) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java]: Got it

**Review by cgardens — APPROVED** [review]

**Pull request merged**